openedx/core/lib/hash_utils.py

```diff
@@ -32,3 +35,32 @@ def short_token():
     Currently, this is just meant to create sufficiently random tokens
     """
     return create_hash256(max_length=32)
+
+
+def hash_usage_key(usage_key: UsageKey) -> str:
+    """
+    Get the blake2b hash key for the given usage_key and encode the value. The
+    hash key will be added to the usage key's mapping dictionary for decoding
+    in LMS.
+
+    Args:
+        usage_key: the id of the location to which to generate the path
+
+    Returns:
+        The string of the encoded hashed key.
+    """
+    short_key = hashlib.blake2b(bytes(str(usage_key), 'utf-8'), digest_size=6)
+    encoded_hash = urlsafe_b64encode(bytes(short_key.hexdigest(), 'utf-8'))
```

Comment:
Minor bit: `short_key` should have a `digest()` method that gets you straight to the bytes, so you don't have to re-convert from the `hexdigest()`, i.e. `urlsafe_b64encode(short_key.digest())` should work I think.